compiler/src/dotty/tools/dotc/core/tasty/TreePickler.scala

```diff
@@ -338,7 +343,12 @@ class TreePickler(pickler: TastyPickler) {
           case _: Template | _: Hole => pickleTree(tpt)
           case _ if tpt.isType => pickleTpt(tpt)
         }
-        pickleTreeUnlessEmpty(rhs)
+        if attributes.isOutline && sym.isTerm && attributes.isJava then
+          // TODO: if we introduce outline typing for Scala definitions
+          // then we will need to update the check here
+          pickleElidedUnlessEmpty(rhs, tpt.tpe)
```

Comment:
For a method we need to use the final result type of `tpt.tpe` here. Otherwise IIUC it's going to be a `MethodicType`, which is not what we want.

Reply:
no for `DefDef` the `tpt` is the tree of the result type, the parameter lists are all in the `paramss` field

Reply:
I'm working on a separate branch where I add a test for this

Reply:
please see https://github.com/lampepfl/dotty/pull/19074